include/range/v3/view/stride.hpp

```diff
@@ -205,10 +205,20 @@ namespace ranges
                         ranges::advance(it, n),
                         ranges::advance(it, n, std::declval<iterator_t<Rng> &>())))
                 {
-                    if(0 == n) return;
+                    if(0 == n)
+                        return;
                     n *= rng_->stride_;
                     auto const last = ranges::end(rng_->base());
-                    if(it == last) n -= rng_->get_offset();
+                    if(it == last)
+                    {
+                        RANGES_EXPECT(n < 0);
+                        if(rng_->get_offset(false) < 0) // hasn't been set yet!
+                        {
+                            auto const rem = ranges::distance(rng_->base()) % rng_->stride_;
```

Comment:
Ditto on line 179.

Reply:
For us to reach this code (advancing the end iterator), we must be adapting a random access, bounded range. So yes, this distance call is O(1).

Reply:
Ok, not exactly "ditto", since 179 doesn't know `SizedSentinel`.  In either case we shouldn't be calling `distance` on a non-`SizedRange` in an O(1) iterator operation. Consequently `stride_view<R>` cannot be `Common` unless `SizedRange<R>`?

Reply:
> For us to reach this code (advancing the end iterator), we must be adapting a random access, bounded range

A random access, common range is always sized; it should have stored the offset in the `stride_view_base<R, true>` constructor.

Reply:
Couldn't we also get here by magically advancing an iterator of a non-`Common` random access range by exactly the number of elements in the range?

Reply:
This function _can_ be called for non-`Common`, non-`Sized`, random access ranges, but in that case, we can only be decrementing the end iterator if that position was found by first advancing the begin iterator to the end, in which case the `offset` would have been set when we reached the end.

AFAICT, the call the `distance` is correct and necessary.